### PR TITLE
(GH-583) Change build script githubref and optimize build times

### PR DESCRIPTION
**Pull request by jpogran**
- (GH-583) Change build script to github format
- (maint) Optimize build script network calls

### CODE REVIEW

**Comment by jpogran** [comment]
Tagging @glennsarti to make sure it does more than work on my machine, even thought CI passed :)

File: psakefile.ps1 [hunk 1]
```diff
@@ -103,9 +113,15 @@ task Bump {
...
 }
 
+task Npm -precondition { !(Test-Path (Join-Path $PSScriptRoot 'node_modules')) }{
+  exec { npm install }
+}
+
 task Vendor -depends VendorEditorServices, VendorEditorSyntax, VendorCytoscape
 
-task Build -depends Clean, Vendor, CompileTypeScript
+task BuildOrig -depends Clean, Npm, Vendor, CompileTypeScript
```
Comment (glennsarti) [inline psakefile.ps1:122]: Do we need to keep BuildOrig?

File: psakefile.ps1 [hunk 2]
```diff
@@ -103,9 +113,15 @@ task Bump {
...
+
+task Build -depends Npm, Vendor, CompileTypeScript
 
 task Initial -depends Clean, Vendor
```
Comment (glennsarti) [inline psakefile.ps1:126]: Do we need to keep Initial given that it's not called in project.json anymore?

File: psakefile.ps1 [hunk 3]
```diff
@@ -16,33 +16,43 @@ task Clean {
...
-    $githubrepo = 'puppet-editor-services'
+# "editorServices": {
+#   "release": "0.26.0"
+# }
+# "editorServices": {
+#   "release": "0.26.0"
+#   "githubrepo": "puppet-editor-services",
+#   "githubuser": "glennsarti"
+# }
+  # "editorServices": {
```
Comment (glennsarti) [inline psakefile.ps1:27]: nit:  Bad indent.

File: psakefile.ps1 [hunk 4]
```diff
@@ -16,33 +16,43 @@ task Clean {
...
+      throw "Invalid githubref. Must be in user:branch format like glennsarti:spike-rearch-langserver"
+    }
+    $githubuser = $githubref.split(":")[0]
+    $githubbranch = $githubref.split(":")[1]
+    $uri = "https://github.com/${githubuser}/${githubrepo}/archive/${githubbranch}.zip"
   }
 
   if ($config.editorComponents.editorServices.directory) {
     Copy-Item -Path $config.editorComponents.editorServices.directory -Destination $languageServerPath -Recurse -Force
   }elseif ($config.editorComponents.editorServices.release) {
```
Comment (glennsarti) [inline psakefile.ps1:50]: nit: Inconsistent bracket style.  I know this isn't part of this PR and I'd be happy to merge it even if not fixed.

**Comment by glennsarti** [comment]
Code Works.  Some minor comments.

File: psakefile.ps1 (hunk 4 above)
Comment (jpogran) [reply psakefile.ps1]: Fixed.

File: psakefile.ps1 (hunk 2 above)
Comment (jpogran) [reply psakefile.ps1]: Thanks. I had kept it during testing with intention to remove

File: psakefile.ps1 (hunk 1 above)
Comment (jpogran) [reply psakefile.ps1]: Thanks. I had kept it during testing with intention to remove

**Comment by jpogran** [comment]
I also fixed up editorSyntax, as it still had lingua-pupuli in it.

**Pull request merged**